fabric-rendering-v1/src/client/java/net/fabricmc/fabric/impl/client/rendering/hud/HudStatusBarHeightRegistryImpl.java

```diff
@@ -0,0 +1,394 @@
+import net.fabricmc.fabric.api.client.rendering.v1.hud.HudElement;
+import net.fabricmc.fabric.api.client.rendering.v1.hud.HudElementRegistry;
+import net.fabricmc.fabric.api.client.rendering.v1.hud.StatusBarHeightProvider;
+import net.fabricmc.fabric.api.client.rendering.v1.hud.VanillaHudElements;
+import net.fabricmc.fabric.mixin.client.rendering.InGameHudAccessor;
+
+public final class HudStatusBarHeightRegistryImpl implements ClientModInitializer {
+	/**
+	 * The height at which vanilla begins rendering status bars; this is used for health and food / mount health.
+	 */
+	static final int DEFAULT_HEIGHT = 39;
+	/**
+	 * The height at which the held item tooltip renders in vanilla; for our purposes we already subtract the default
+	 * height.
+	 */
+	static final int HELD_ITEM_TOOLTIP_HEIGHT = 59 - DEFAULT_HEIGHT;
+	/**
+	 * The height at which the overlay message (from playing records, or unsuccessfully trying to sleep) renders in
+	 * vanilla; for our purposes we already subtract the default height.
+	 */
+	static final int OVERLAY_MESSAGE_HEIGHT = 68 - DEFAULT_HEIGHT;
+	static final int TEXT_HEIGHT_DELTA = OVERLAY_MESSAGE_HEIGHT - HELD_ITEM_TOOLTIP_HEIGHT;
+	/**
+	 * Height provider for the vanilla health bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addLeft(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider HEALTH_BAR = (PlayerEntity player) -> {
+		InGameHud hud = MinecraftClient.getInstance().inGameHud;
+		int playerHealth = MathHelper.ceil(player.getHealth());
+		int displayHealth = ((InGameHudAccessor) hud).getRenderHealthValue();
+		float maxHealth = Math.max((float) player.getAttributeValue(EntityAttributes.MAX_HEALTH),
+				Math.max(displayHealth, playerHealth));
+		int absorptionAmount = MathHelper.ceil(player.getAbsorptionAmount());
+		int healthRows = MathHelper.ceil((maxHealth + absorptionAmount) / 2.0F / 10.0F);
+		int rowShift = Math.max(10 - (healthRows - 2), 3);
+		return 10 + (healthRows - 1) * rowShift;
+	};
+	/**
+	 * Height provider for the vanilla armor bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addLeft(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider ARMOR_BAR = (PlayerEntity player) -> {
+		return player.getArmor() > 0 ? 10 : 0;
+	};
+	/**
+	 * Height provider for the vanilla mount health.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
```

Comment:
I don't think this paragraph is needed on each of these fields, especially since this is an impl class.